editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/preferences/StagingRepositoryListEditor.java

```diff
@@ -0,0 +1,115 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.fusesource.ide.projecttemplates.preferences;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringTokenizer;
+
+import org.eclipse.jface.preference.ListEditor;
+import org.eclipse.jface.window.Window;
+import org.eclipse.swt.widgets.Composite;
+
+/**
+ * Field editor for the list of staging repositories. Based on the PathEditor in
+ * Eclipse.
+ * 
+ * @author brianf
+ */
+public class StagingRepositoryListEditor extends ListEditor {
+
+	/**
+	 * The last repository URL specified, or <code>null</code> if none.
+	 */
+	private String lastURL;
+
+	/**
+	 * The last repository name specified, or <code>null</code> if none.
+	 */
+	private String lastName;
+
+	/**
+	 * Creates a new repository list field editor
+	 */
+	protected StagingRepositoryListEditor() {
+	}
+
+	/**
+	 * Creates a staging repository field editor.
+	 *
+	 * @param name
+	 *            the name of the preference this field editor works on
+	 * @param labelText
+	 *            the label text of the field editor
+	 * @param parent
+	 *            the parent of the field editor's control
+	 */
+	public StagingRepositoryListEditor(String name, String labelText, Composite parent) {
+		init(name, labelText);
+		createControl(parent);
+	}
+
+	@Override
+	protected String createList(String[] items) {
+		StringBuffer repositories = new StringBuffer("");//$NON-NLS-1$
+
+		for (int i = 0; i < items.length; i++) {
+			repositories.append(items[i]);
+			repositories.append(StagingRepositoriesConstants.REPO_SEPARATOR);
+		}
+		return repositories.toString();
+	}
+	
+	public String getItemList() {
+		return createList(super.getList().getItems());
+	}
+	
+	public boolean isListEnabled() {
+		return getList().isEnabled();
+	}
+
+	@Override
+	protected String getNewInputObject() {
+
+		StagingRepositoryDialog dialog = new StagingRepositoryDialog(getShell());
+		if (lastName != null) {
+			dialog.setName(lastName);
+		}
+		if (lastURL != null) {
+			dialog.setURL(lastURL);
+		}
+		int rtn = dialog.open();
+		if (rtn == Window.OK && dialog.getURL() != null && dialog.getName() != null) {
+			String outUrl = dialog.getURL().trim();
+			String outName = dialog.getName().trim();
+			if (outUrl.length() == 0 || outName.length() == 0) {
+				return null;
+			}
+			lastName = outName;
+			lastURL = outUrl;
+			return outName + StagingRepositoriesConstants.NAME_URL_SEPARATOR + outUrl;
```

Comment:
I think that it implies to write a full custom preference page then (and throw almost all of this PR) but maybe I'm wrong. Perhaps Brian who worked on it has an idea which doesn't require it.
We need to keep in mind that these parameters are targeting mostly internal users who are using versions not yet released.